azure-pipelines.yml

```diff
@@ -467,6 +467,35 @@ jobs:
     parameters:
       includePatterns:
         - "/hw/***"
+  - task: DownloadSecureFile@1
+    name: gcpkey
+    inputs:
+      secureFile: gcpkey.json
+  - bash: |
+      echo "Installing gsutil"
+      sudo apt-get install -y apt-transport-https ca-certificates gnupg
+      echo "deb [signed-by=/usr/share/keyrings/cloud.google.gpg] https://packages.cloud.google.com/apt cloud-sdk main" \
+        | sudo tee -a /etc/apt/sources.list.d/google-cloud-sdk.list
+      curl https://packages.cloud.google.com/apt/doc/apt-key.gpg | sudo tee /usr/share/keyrings/cloud.google.gpg
+      sudo apt-get update || {
+        error "Failed to run apt-get update"
+      }
+      sudo apt-get install -y google-cloud-cli
+
+      date -u > latest.txt
+      echo $(Build.SourceVersion) >> latest.txt
```

Comment:
I would compose this into one command...
```sh
echo "$(date -u +%Y-%m-%dT%H:%M:%S)-$(Build.SourceVersion)" > latest.txt
```
It's a bit annoyiing that date insert a newline, and to my knowledge it can't be avoided. You could just use `tr -d '\n'`, I guess.